chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -519,6 +519,30 @@ impl<'a, S: TransactionVerifierStorageRef> TransactionVerifier<'a, S> {
             .undo
     }
 
+    fn get_tx_cache_ref(&self) -> Result<Option<&TxIndexCache>, ConnectTransactionError> {
+        let enabled = self
+            .storage_ref
+            .get_is_mainchain_tx_index_enabled()?
```

Comment:
I don't think this is necessary. The way I imagine it, is that the flag is checked for consistency when chainstate is constructed, and then whether tx index is enabled is passed as configuration to TransactionVerifier. Maybe there should be a TransactionVerifier configuration object. In fact, we do need that for the mempool.

Reply:
I added new `TransactionVerifierConfig` struct which is passed to `TransactionVerifier`. Will that work?

I also added `allow(clippy::too_many_arguments)` in two places to suppress clippy warnings.